GMLWriter/src/com/github/systemdir/gml/YedGmlWriter.java

```diff
@@ -92,62 +93,74 @@ public Builder(YedGmlGraphicsProvider<V1, E1, G1> graphProvider, PrintLabels...
          */
-        public Builder<V1, E1, G1> setVertexIDProvider(VertexNameProvider vertexIDProvider) {
+        public Builder<V1, E1, G1> setVertexIDProvider(Function<Object, String> vertexIDProvider) {
             this.vertexIDProvider = vertexIDProvider;
             return this;
         }
 
         /**
-         * For generating vertex labels. If null, vertex labels will be generated using the toString() method of the vertex object.
+         * For generating vertex labels.
+         * <p>
+         * If null, vertex labels will be generated using the toString() method of the vertex object.
          *
-         * @param vertexLabelProvider
+         * @param vertexLabelProvider The function must not return null
+         * @return this
          */
-        public Builder<V1, E1, G1> setVertexLabelProvider(VertexNameProvider<V1> vertexLabelProvider) {
+        public Builder<V1, E1, G1> setVertexLabelProvider(Function<V1, String> vertexLabelProvider) {
             this.vertexLabelProvider = vertexLabelProvider;
             return this;
         }
 
         /**
-         * Sets a id provider for the vertexes. This will default to {@link org.jgrapht.ext.IntegerEdgeNameProvider} if not set.
+         * Sets a id provider for the vertexes.
+         * <p>
+         * This will default to {@link com.github.systemdir.gml.model.UniqueIntIdFunction} if not set.
          *
-         * @param edgeIDProvider id provider for the vertexes to use.
+         * @param edgeIDProvider The function must not return null
+         * @return this
          */
-        public Builder<V1, E1, G1> setEdgeIDProvider(EdgeNameProvider<E1> edgeIDProvider) {
+        public Builder<V1, E1, G1> setEdgeIDProvider(Function<E1, String> edgeIDProvider) {
             this.edgeIDProvider = edgeIDProvider;
             return this;
         }
 
         /**
-         * For generating edge labels. If null, edge labels will be generated using the toString() method of the edge object.
+         * For generating edge labels.
+         * <p>
+         * If null, edge labels will be generated using the toString() method of the edge object.
          *
-         * @param edgeLabelProvider
+         * @param edgeLabelProvider The function must not return null
+         * @return this
          */
-        public Builder<V1, E1, G1> setEdgeLabelProvider(EdgeNameProvider<E1> edgeLabelProvider) {
+        public Builder<V1, E1, G1> setEdgeLabelProvider(Function<E1, String> edgeLabelProvider) {
             this.edgeLabelProvider = edgeLabelProvider;
             return this;
         }
 
         public YedGmlWriter<V1, E1, G1> build() {
-            return new YedGmlWriter<V1, E1, G1>(this);
+            return new YedGmlWriter<>(this);
         }
     }
 
 
-    private static final String creator = "JGraphT GML Exporter - modified by Hayato Hess";
+    private static final String creator = "JGraphT GML Exporter - modified by Hayato Hess - modified by Andreas Hofstadler";
```

Comment:
I would suggest something like "JGraphT GML Exporter - modified by Hayato Hess, Andreas Hofstadler" if you don't mind.